src/gate/parserRIL/main.cpp

```diff
@@ -0,0 +1,406 @@
+//      return;
+//    }
+//    a = connAsWire;
+//    symbol = logicFunction(cellSec->type.index_);
+//    typeFunc.emplace(a, symbol);
+//    if (symbol == "~") {
+//      flag = 1;
+//    }
+//    if (++it == conns.end()) {
+//      return;
+//    }
+//    if (!flag) {
+//      b = connAsWire;
+//      flag = 0;
+//    } else {
+//      b = connAsWire;
+//      c = b;
+//      cell.emplace(a, std::make_pair(b, c));
+//    }
+//    if (++it == conns.end()) {
+//      return;
+//    }
+//    c = connAsWire;
+//    cell.emplace(a, std::make_pair(b, c));
+//    if (++it == conns.end()) {
+//      return;
+//    }
+//    a = connAsWire;
+//    if (++it == conns.end()) {
+//      return;
+//    }
+//    b = connAsWire;
+//    cell.emplace(c, std::make_pair(a, b));
+//    typeFunc.emplace(c, symbol);
+
+//}
+
+void printCells(const YLib::dict<RTlil::IdString, RTlil::Cell *> &cells,
+                std::ofstream &fout,
+                std::map<int, std::pair<int, int>> &cell,
+                std::map<int, std::string> &typeFunc) {
+
+  for (auto it1 = cells.begin(); it1 != cells.end(); ++it1) {
+      size_t i = 0;
+      bool flag = 0;
+      std::string symbol;
+      int a, b, c;
+    for (auto it2 = it1->second->connections_.begin();
+         it2 != it1->second->connections_.end(); ++it2) {
+
+      auto connAsWire = it2->second.as_wire()->name.index_;
+      //fillCells(cell, typeFunc, connAsWire, it1->second);
+
+      i++;
+      if (i == 1) {
+        a = connAsWire;
+        symbol = logicFunction(it1->second->type.index_);
+        typeFunc.emplace(a, symbol);
+        if (symbol == "~") {
+          flag = 1;
+        }
+      }
+      if (i == 2) {
+        if (!flag) {
+          b = connAsWire;
+          flag = 0;
+        } else {
+          b = connAsWire;
+          c = b;
+          cell.emplace(a, std::make_pair(b, c));
+        }
+      }
+      if (i == 3) {
+        c = connAsWire;
+        cell.emplace(a, std::make_pair(b, c));
+      }
+      if (i == 4) {
+        a = connAsWire;
+      }
+      if (i == 5) {
+        b = connAsWire;
+        cell.emplace(c, std::make_pair(a, b));
+        typeFunc.emplace(c, symbol);
+      }
+    }
+  }
+
+}
+
+void printConnections(
+    const std::vector<std::pair<RTlil::SigSpec, RTlil::SigSpec>> &connections,
+    std::ofstream &fout,
+    std::map<int, std::pair<int, int>> &cell,
+    std::map<int, std::string> &inputs, std::map<int, std::string> &outputs,
+    std::map<int, std::string> &typeFunc) {
+  std::ostringstream buf;
+  for (auto it1 = connections.begin(); it1 != connections.end(); ++it1) {
+    auto connAsWireInput = it1->second.as_wire()->name.index_;
+    auto connAsWireOutput = it1->first.as_wire()->name.index_;
+    if (inputs.find(connAsWireInput) == inputs.end()) {
+      fout << "@(*) {\n";
+      fout << "   " << outputs.find(connAsWireOutput)->second
+           << " = " << buildRIL(buf, connAsWireInput, cell, inputs, typeFunc) << ";\n";
+      fout << "}\n";

```

Comment:
No need for a buffer if you write to the output directly.
```suggestion
      fout << "@(*) {\n";
      fout << "   " << outputs.find(connAsWireOutput)->second << " = ";
      buildRIL(fout, connAsWireInput, cell, inputs, typeFunc);
      fout << ";\n" << "}\n";

```